XMonad/Layout/IndependentScreens.hs

```diff
@@ -171,17 +191,36 @@ whenCurrentOn s pp = pp
         _ -> ppOutput pp out
     }
 
--- | If @vSort@ is a function that sorts 'WindowSpace's with virtual names, then @marshallSort s vSort@ is a function which sorts 'WindowSpace's with physical names in an analogous way -- but keeps only the spaces on screen @s@.
-marshallSort :: ScreenId -> ([WindowSpace] -> [WindowSpace]) -> ([WindowSpace] -> [WindowSpace])
+-- | Filter workspaces that are on current screen.
+workspacesOn :: ScreenId -> [PhysicalWindowSpace] -> [PhysicalWindowSpace]
+workspacesOn s = filter (\ws -> unmarshallS (W.tag ws) == s)
+
+-- | @vSort@ is a function that sorts 'VirtualWindowSpace's with virtual names.
+-- @marshallSort s vSort@ is a function which sorts 'PhysicalWindowSpace's with virtual names,
+-- but keeps only the 'WindowSpace'\'s on screen @s@.
+--
+-- NOTE: This function always sorts the workspace names lexically, see
+-- <https://github.com/xmonad/xmonad-contrib/issues/420 this issue>.
+-- Only use it if you want that behaviour!
+marshallSort :: ScreenId -> ([VirtualWindowSpace] -> [VirtualWindowSpace]) -> ([PhysicalWindowSpace] -> [PhysicalWindowSpace])
 marshallSort s vSort = pScreens . vSort . vScreens where
-    onScreen ws = unmarshallS (tag ws) == s
-    vScreens    = map unmarshallWindowSpace . filter onScreen
+    vScreens    = map unmarshallWindowSpace . workspacesOn s
     pScreens    = map (marshallWindowSpace s)
 
+-- | To still sort by 'VirtualWorkspaces', you can use 'marshallSort' like this:
```

Comment:
This comment still won't be rendered by haddock.

Reply:
Oh, must have missed that.  I'll move it over to the `marshallSort` docs